### DISCUSSION TITLE
impossible to connect a domain

### DISCUSSION THREAD

**Discussion by bensz** [the-djmaze/snappymail#817] [Q&A]
Hello,
I want to use snappymail in my nextcloud.
My problem is I can't create a domain. In administration UI, I sent my domain name, imap and smtp address. When I try test, I have no errors message, but that doesn't works.

I record the log, attached, but I don't understand array error. Connection seems to be right.

Thanks for help
Benoit
[log.txt](https://github.com/the-djmaze/snappymail/files/10327482/log.txt)

**Comment by MI-KY** [comment]
It seams your username or password could be incorrect.
As you log in only with a username and not a complete mail address maybe the wrong domain is used or you have to setup the domain config to use short usernames... Have a look at: https://github.com/the-djmaze/snappymail/wiki/Admin-Manual#menu-domains.
An other error could raise if your default domain is not the correct one (Menu "Login" in Admin Panel).

**Reply by bensz** [reply]
Hello and happy new year,
Thanks for your reply.
I looked this doc and I used it. For the username and password, I use them in my phone to connect my mailbox. So I'm sure of them.
I mustn't use short username, my provider use my email address as username. And there's no risk of using wrong domain as I can't create the first. This log is from test button in "add domain". When I click add in this box, a little message appear "the request expire" under the domain name.

For info, I use nextcloud 25.01 and snappy 2.24.4

Thanks,
Benoit

**Reply by MI-KY** [reply]
Hi, happy new year!
Oh, ok I understand - maybe @the-djmaze can help you with this details - I first don't saw the Socket error at the beginning of the log...

You maybe have already made sure that from your Nextcloud server the imap server is reachable (out of the shell)?

**Reply by bensz** [reply]
I installed the default mail app in nextcloud and it worked.
Do you think @the-djmaze will read this discussion or is it better to use another way?
Thanks
Benoit

**Reply by MI-KY** [reply]
Ok perfect 👍 .
Normally the-djmaze reads them 🙂

**Reply by bensz** [reply]
thanks a lot for your help
Benoit

**Reply by the-djmaze** [reply]
I read, but need PC to analyze the log.
Stay tuned

**Comment by the-djmaze** [comment]
Your test uses:
```
IMAP[INFO]: Start connection to "ssl://mail.infomaniak.com:993"
```

Your login uses:
```
IMAP[INFO]: Start connection to "tcp://mail.infomaniak.com:993"
IMAP[ERROR]: Stream Meta: Array
(
    [timed_out] => 
    [blocked] => 1
    [eof] => 1
    [stream_type] => tcp_socket/ssl
    [mode] => r+
    [unread_bytes] => 0
    [seekable] => 
)
```

Somehow the Domain setup is not using SSL.
Also the SMTP says the login is wrong.

Best to disable all Domains in admin and activate only one for testing as it looks like you have multiple domains that are conflicting (the wildcard probably?)

**Comment by bensz** [comment]
Hello and thanks for your reply.
I don't have configured domains. I try to configure the first. 
In the first attach, you'll see my imap conf. On the second, my smtp conf, on the third, the return when I click add domain.

For the login/ password and the other parameters, they came from my phone which is connected to my mail.
The only thing, is in infomaniak I have a 2 authentications. I receive an SMS. Can snappy be blocked for this reason?
Thanks
Benoit

![Capture d’écran du 2023-01-02 21-42-09](https://user-images.githubusercontent.com/11018363/210276805-91199187-006e-4bc9-b15d-e1ba3f03cdfd.png)
![Capture d’écran du 2023-01-02 21-42-48](https://user-images.githubusercontent.com/11018363/210276806-19fe1c21-82a5-4271-8d3e-c80348cb6fa2.png)
![Capture d’écran du 2023-01-02 21-44-35](https://user-images.githubusercontent.com/11018363/210276807-a082e5a2-f55c-416e-a0c9-214dbd9367a5.png)

**Reply by bensz** [reply]
Hello,
For information, I upgrade to v 2.24.5, but there's no change

**Comment by bensz** [comment]
Hello,
I upgrade but no change.
Do you have any ideas about where to search?
Thanks
Benoit

**Comment by the-djmaze** [comment]
Is your firewall blocking outgoing connections?
